library/x509_crt.c

```diff
@@ -1403,10 +1404,39 @@ int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const char *path )
             ret = MBEDTLS_ERR_X509_BUFFER_TOO_SMALL;
             goto cleanup;
         }
-        else if( stat( entry_name, &sb ) == -1 )
+        else
         {
-            ret = MBEDTLS_ERR_X509_FILE_IO_ERROR;
-            goto cleanup;
+            /* determinate if the file entry could be a link, using lstat(2)
```

Comment:
Minor: `determine` instead of `determinate`?

Or, alternatively, a suggestion for the entire comment: "Use lstat instead of stat to not fail in case of a broken link. In case of a valid link, another call to stat is needed below to query the properties of the file the link points to."